### DISCUSSION TITLE
How to log when the program close itself at launch?

### DISCUSSION THREAD

**Discussion by Unknown78** [woelper/oculante#244] [Q&A] [answered]
I'm on Windows 11.

I want to create an issue where oculante just close itself at launch.
The problem is that I don't know how to log it.

There is nothing in the Event Viewer or `%LOCALAPPDATA%\CrashDumps`
In Android we have `adb logcat`, but in Windows what do I have to do?

**Comment by woelper** [comment, marked as answer]
Hi! First of all thanks for spotting this and helping to improve Oculante!

Oculante logs to stdout by default, so if you start the program in a terminal, you should see something.

I am using env_logger in oculante, which sets log levels based on the RUST_LOG enviroment variable. There is more background info here:
https://docs.rs/env_logger/latest/env_logger/

basically, you would have to set RUST_LOG to debug in the terminal and then start Oculante - for example `set RUST_LOG=debug` and then `oculante.exe`. Please let me know if I can help of if you find out more!

**Reply by Unknown78** [reply]
Hi, you're welcome.

I've tried it with `set RUST_LOG=debug && oculante > oculante.log`
And it outputs nothing at all.
It turns out you also need to combine it with `stderr`.
`set RUST_LOG=debug && oculante > oculante.log 2>&1`

I will open an issue.
Edit: this is the issue: https://github.com/woelper/oculante/issues/247

<hr>

These precious knowledge should be available to everyone easily.
I've seen you've had hundreds of issues, this would help newcomers to log any issues better.
And I am strongly suggests that this should be on the topmost of `README.md` of the Code section.

**Reply by woelper** [reply]
Good points!
I think this should be in the readme as well. I also think that logging should just work - at least errors should be directly output to the terminal. Maybe there is something special about windows. On linux I believe there is nothing one has to set in order to see errors at least.

In terms of your issue, it seems that your graphics card might miss capabilities that the engine I use requires. In that case there probably is not much I can do sadly.
